Feature/webrdfs create ls

heliumj opened the pull request:
Add webRDFS handlers for create and ls. Also added tests for both

heliumj committed:
add ls and create handlers

heliumj committed:
add test for create and ls

heliumj committed:
fix lint

vnazz reviewed (commented):
Change your tests to use op=CREATE and op= LISTSTATUS because right now they are just testing read

vnazz commented:
Also add some routing in the get_handler so it calls the create_file_handler and the ls_handler based on the url

Commit:
caching for ZkWrapper->get (#180)

* added initial cache stuff

* lru cache file

* fix indentation

* lru cache cmake

* added cache code to get

* finished cache code for zk wrapper get

* cache watch

* fixed include errors

* fixed linking errors

* fixed some tests and added more logging

* more testing logs

* close zk on teardown

* fixed all tests

* lint errors and rebase

heliumj committed:
fix handler

vnazz commented on web-rdfs/source/http_handlers.cc:
```diff
@@ -77,25 +95,25 @@ void get_handler(std::shared_ptr<HttpsServer::Response> response,
+    create_file_handler(response, path);
+  }
+  else {
+    ls_handler(response, request);
```
Might be better to make this an else if just so you make sure ls only occurs on LISTSTATUS

vnazz commented on test/webRDFS/WebServerTest.cc:
```diff
@@ -45,6 +45,41 @@ TEST(WebServerTest, testRead) {
+      // Check that results match
+      ASSERT_EQ(0,
+                system("diff /home/vagrant/rdfs/test/webRDFS/"
+                               "expectedResultRead actualResultRead"));
```
Also change the name of the actualResultRead / expectedResultRead to actualResultLs and expectedResultLs because as of now the test will fail because the result of a read isn't the same as ls.

vnazz commented on test/webRDFS/WebServerTest.cc:
```diff
@@ -45,6 +45,41 @@ TEST(WebServerTest, testRead) {
+
+      ASSERT_EQ(0,
+                system("curl -i --insecure https://localhost:8080/webhdfs/v1/"
+                               "fileToRead?op=CREATE > actualResultRead"));
```
Also change the name of the actualResultRead / expectedResultRead to actualResultCreate and expectedResultCreate because as of now the test will fail because the result of a read isn't the same as create

Commit:
Feature/ec (#186)

* Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

blockIndices optional field added for EC purpose.

a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

Feature/EC-ISAL adds ISA-L into RDFS (#121)

* Addition of ISA-L wrapper.

* Modified CMake to build ISAL wrapper correctly.

* Added ErasureCodeTest files.

* Added scripts to pass travis build checks.

* Added tests. Also added scripts for travis build checks.

* Modified vagrant_provision to make sure vagrant is set up correctly.

* Deleted unnecessary commented line in tests.

* Changed links for isal.tar.gz to the one on Rice Server.

* chmod +x isal.sh

* Fixed minor spelling error present.

* Added making of the ISAL library into .ci/install/isal.sh

new proto messages related to a new set of EC RPC calls. (#122)

Fixed error with vagrant provisioning! (#137)

* Fixed error with vagrant provisioning!

* Fixed an error in travis building when downloading the ISAL.

a new set of NameNode RPC APIs (#123)

* an namenode api that returns the set of registered EC policies implemented.

* register getErasureCodingPolicies in the translator

Hns test fix (#139)

* Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

* blockIndices optional field added for EC purpose.

* a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

* remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

* Feature/EC-ISAL adds ISA-L into RDFS (#121)

* Addition of ISA-L wrapper.

* Modified CMake to build ISAL wrapper correctly.

* Added ErasureCodeTest files.

* Added scripts to pass travis build checks.

* Added tests. Also added scripts for travis build checks.

* Modified vagrant_provision to make sure vagrant is set up correctly.

* Deleted unnecessary commented line in tests.

* Changed links for isal.tar.gz to the one on Rice Server.

* chmod +x isal.sh

* Fixed minor spelling error present.

* Added making of the ISAL library into .ci/install/isal.sh

* new proto messages related to a new set of EC RPC calls. (#122)

* Fixed error with vagrant provisioning! (#137)

* Fixed error with vagrant provisioning!

* Fixed an error in travis building when downloading the ISAL.

* a new set of NameNode RPC APIs (#123)

* an namenode api that returns the set of registered EC policies implemented.

* register getErasureCodingPolicies in the translator

* hopefully fixed branhc

* can't find get_num_data_parity_blocks, everything else builds

* fixed lint error

* test passes

* fixed lint error

* another lint error

added isEC field to the FileZNode. (#149)

Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

blockIndices optional field added for EC purpose.

a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

new proto messages related to a new set of EC RPC calls. (#122)

Fixed error with vagrant provisioning! (#137)

* Fixed error with vagrant provisioning!

* Fixed an error in travis building when downloading the ISAL.

a new set of NameNode RPC APIs (#123)

* an namenode api that returns the set of registered EC policies implemented.

* register getErasureCodingPolicies in the translator

Hns test fix (#139)

* Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

* blockIndices optional field added for EC purpose.

* a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

* remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

* Feature/EC-ISAL adds ISA-L into RDFS (#121)

* Addition of ISA-L wrapper.

* Modified CMake to build ISAL wrapper correctly.

* Added ErasureCodeTest files.

* Added scripts to pass travis build checks.

* Added tests. Also added scripts for travis build checks.

* Modified vagrant_provision to make sure vagrant is set up correctly.

* Deleted unnecessary commented line in tests.

* Changed links for isal.tar.gz to the one on Rice Server.

* chmod +x isal.sh

* Fixed minor spelling error present.

* Added making of the ISAL library into .ci/install/isal.sh

* new proto messages related to a new set of EC RPC calls. (#122)

* Fixed error with vagrant provisioning! (#137)

* Fixed error with vagrant provisioning!

* Fixed an error in travis building when downloading the ISAL.

* a new set of NameNode RPC APIs (#123)

* an namenode api that returns the set of registered EC policies implemented.

* register getErasureCodingPolicies in the translator

* hopefully fixed branhc

* can't find get_num_data_parity_blocks, everything else builds

* fixed lint error

* test passes

* fixed lint error

* another lint error

added isEC field to the FileZNode. (#149)

Feature/EC_Worker (#154)

* Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

* blockIndices optional field added for EC purpose.

* a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

* remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

* Feature/EC-ISAL adds ISA-L into RDFS (#121)

* Addition of ISA-L wrapper.

* Modified CMake to build ISAL wrapper correctly.

* Added ErasureCodeTest files.

* Added scripts to pass travis build checks.

* Added tests. Also added scripts for travis build checks.

* Modified vagrant_provision to make sure vagrant is set up correctly.

* Deleted unnecessary commented line in tests.

* Changed links for isal.tar.gz to the one on Rice Server.

* chmod +x isal.sh

* Fixed minor spelling error present.

* Added making of the ISAL library into .ci/install/isal.sh

* new proto messages related to a new set of EC RPC calls. (#122)

* Added constants to zk_client_common for new ec_recover workqueue.

* Fixed typos.

* very in progress

* Added constants to zk_client_common for new ec_recover workqueue.

* Fixed typos.

* very in progress

* Watches health of datanodes and added functionality to post ec_recover
jobs.

* Fixed error of not casting block_id received from zk to uint64_t

* filled out handleReconstructCmds

* minor fixes

* started daemon thread in dn main.cc

* Added check_acks functionality for EC

* Fixed build errors.

* made remote reads modular

* fixed build errors

* reconstruction reads building

* added writeBlock method

* Added recovery computations.

* Made lint changes

* Added constants to zk_client_common for new ec_recover workqueue.

* Fixed typos.

* Added constants to zk_client_common for new ec_recover workqueue.

* very in progress

* Watches health of datanodes and added functionality to post ec_recover
jobs.

* Fixed error of not casting block_id received from zk to uint64_t

* filled out handleReconstructCmds

* minor fixes

* started daemon thread in dn main.cc

* Added check_acks functionality for EC

* Fixed build errors.

* made remote reads modular

* fixed build errors

* reconstruction reads building

* added writeBlock method

* Added recovery computations.

* Made lint changes

* Fixing lint.

added setPolicy proto (#166)

fixed block release

add block completed. (#151)

* add block completed.

* fixes the block generation id test.

* changes made based on comments.

* unit test added and bugs fixed.

* add block completed.

* fixes the block generation id test.

* changes made based on comments.

* unit test added and bugs fixed.

* add one more test and changes that populate the blockindices field.

* building, but tests fail

* fixed merge error

* fixed lint errors

add get ec policy of a path api and unit tests (#177)

* add get ec policy of a path api and unit tests

* set erasurecoindg policy implemented.

* set a src in the unit test.

added ec logic to blockReceived (#184)

* merging

* lint fixes

heliumj committed:
fixing tests

heliumj committed:
fixing tests

vnazz committed:
Feature/move web request translator (#191)

* Move web request translator

* Fix lint errors

* Fix more lint issues

ycxuyingchen committed:
Feature/webrdfs rename mkdir (#189)

rename and mkdir handlers for webrdfs

willkoh committed:
Correctly checks given ecpolicy_name to set if file is ec or not. (#192)

Commit:
ZkNnClient log cleanup (#190)

* changed some todos

* fixed logs to state method and removed extraneous code

* forgot a parenthesis

* missed one error log that should be an info log

* fixed lint errors

Commit:
small change to rebuild (#193)

LINKIWI commented on test/webRDFS/expectedResultCreate:
```diff
@@ -0,0 +1,4 @@
+HTTP/1.1 200 OK
+Content-Length: 13
+
+/fileToCreate
```
This doesn't seem to conform to the [webHDFS API for create](https://hadoop.apache.org/docs/r1.0.4/webhdfs.html#CREATE).

It should return HTTP 201 with a zero content length and an empty response body.

heliumj replied:
Should it also include the location?

LINKIWI replied:
I don't think so; the API spec suggests that the response body is always empty

LINKIWI commented on web-rdfs/source/http_handlers.cc:
```diff
@@ -77,25 +99,24 @@ void get_handler(std::shared_ptr<HttpsServer::Response> response,
+  } else if (!typeOfRequest.compare("LISTSTATUS")) {
+    ls_handler(response, request, path);
   }
 }
```
Since this is `else if`, we should probably have a 404 handler to ensure that every logic path at least causes the server to respond to the client. Probably not necessary to add it in this PR, but I'm tracking in #195.

LINKIWI commented on test/webRDFS/WebServerTest.cc:
```diff
@@ -45,6 +45,44 @@ TEST(WebServerTest, testRead) {
+        "-copyFromLocal fileForTesting /fileToCreate");
+
+      ASSERT_EQ(0,
+                system("curl -i --insecure https://localhost:8080/webhdfs/v1/"
```
`op=CREATE` should not be respected by HTTP GET, according to [spec](https://hadoop.apache.org/docs/r1.0.4/webhdfs.html#CREATE). This should be `curl -X PUT` and the corresponding logic for invoking create should be in `put_handler`

heliumj replied:
fixed

LINKIWI commented on test/webRDFS/WebServerTest.cc:
```diff
@@ -45,6 +45,44 @@ TEST(WebServerTest, testRead) {
+        "-copyFromLocal fileForTesting /fileToCreate");
+
+      ASSERT_EQ(0,
+                system("curl -i --insecure https://localhost:8080/webhdfs/v1/"
```
Do we have server-side logic to handle writing the multipart form data passed by curl via `-T <filename>` to the newly created file?

LINKIWI reviewed (changes requested):
Before you merge--just some small changes to ensure that the `op=CREATE` response is spec-compliant

willkoh committed:
bug/isal (#194)

* Fixed ISA-L building to correctly build.

* Updated link for new isal tar file.

* Updated travis ci install script for isal.

* Fixed typo in isal.sh

* Fixed make instructions.

* Correctly links new library for ISAL

* Properly cleans old versions of isal every provision.

* Changed links for hadoop3 that allows for isa-l to be used by clients.

* Fixed typo in filename for new hadoop3 tar file.

* Fixed more typos.

* Reverted travis hadoop install to previous way of installing if it doesn't exists.

* Changed isal installation for travis builds to not use cache

willkoh committed:
Fixes an uninitialization bug for file_exists. (#197)

cannontwo committed:
Fixed dirlist accumulation (#187)

* Fixed dirlist accumulation

* Fixed dirlist accumulation

* Feature/move web request translator (#191)

* Move web request translator

* Fix lint errors

* Fix more lint issues

* Feature/webrdfs rename mkdir (#189)

rename and mkdir handlers for webrdfs

* Correctly checks given ecpolicy_name to set if file is ec or not. (#192)

* ZkNnClient log cleanup (#190)

* changed some todos

* fixed logs to state method and removed extraneous code

* forgot a parenthesis

* missed one error log that should be an info log

* fixed lint errors

* small change to rebuild (#193)

* bug/isal (#194)

* Fixed ISA-L building to correctly build.

* Updated link for new isal tar file.

* Updated travis ci install script for isal.

* Fixed typo in isal.sh

* Fixed make instructions.

* Correctly links new library for ISAL

* Properly cleans old versions of isal every provision.

* Changed links for hadoop3 that allows for isa-l to be used by clients.

* Fixed typo in filename for new hadoop3 tar file.

* Fixed more typos.

* Reverted travis hadoop install to previous way of installing if it doesn't exists.

* Changed isal installation for travis builds to not use cache

* Fixes an uninitialization bug for file_exists. (#197)

* Fixed dirlist accumulation

willkoh committed:
Removed unnecessary line (#198)

vnazz committed:
Feature/webrdfs api change (#201)

* Fix api issues with webRDFS

* Add destination to rename path

* Add helper functions

LINKIWI committed:
Redirect ISAL compilation output to /dev/null (#202)

LINKIWI committed:
hdfs3/hadoop3 alias, default to v2 (#200)

heliumj committed:
fix test to comply with API

heliumj committed:
fix https response

LINKIWI committed:
[WIP] Add webRDFS SSL cert as root CA (#203)

* Regenerated SSL certs with comp413.local CN

* SSL root CA in CI

* [WIP] Use comp413.local in tests without --insecure

* Use comp413.local in all tests

LINKIWI committed:
Hostname comp413.local as 127.0.0.1 in /etc/hosts (#207)

jessicayu626 committed:
Feature/fix block size error (#205)

* First commit.

* Fix lint errors.

heliumj committed:
fix lint

vnazz commented on web-rdfs/source/http_handlers.cc:
```diff
@@ -11,11 +11,33 @@ void setZk(zkclient::ZkNnClient *zk_arg) {
+
+  zkclient::ZkNnClient::CreateResponse zkResp = zk->create_file(req, res);
+
+  response->write(webRequestTranslator::getCreateResponse(path));
```
since getCreateResponse writes the whole header use *response << webRequestTranslator::getCreateResponse(path)

cannontwo committed:
Feature/metadata consistency (#210)

* Changed flush default argument to false. Operations needing a blocking sync should explicitly pass true.

* Changed flush default argument to false. Operations needing a blocking sync should explicitly pass true.

* Implemented initial conneciton to zookeeper process of record for NameNode connections

* Implemented initial connection to ZooKeeper process of record for datanode

* Changed flush default argument to false. Operations needing a blocking sync should explicitly pass true.

* Implemented initial conneciton to zookeeper process of record for NameNode connections

* Implemented initial connection to ZooKeeper process of record for datanode

* Added edge case handling for concurrent PoR node creation

* Added logic to recover in case the process of record goes down.

* Changed flush default argument to false. Operations needing a blocking sync should explicitly pass true.

* Implemented initial conneciton to zookeeper process of record for NameNode connections

* Implemented initial connection to ZooKeeper process of record for datanode

* Added edge case handling for concurrent PoR node creation

* Added logic to recover in case the process of record goes down.

* Fixed lint errors

* Added helpful TODO for next year's class

LINKIWI committed:
Fix incorrectly named directories in Vagrant provision (#212)

LINKIWI commented on web-rdfs/source/http_handlers.cc:
```diff
@@ -77,25 +99,30 @@ void get_handler(std::shared_ptr<HttpsServer::Response> response,
     read_file_handler(response, path);
-  } else {
-    create_file_handler(response, request);
+  } else if (!typeOfRequest.compare("CREATE")) {
```
Shouldn't this be moved to `put_handler`?

LINKIWI commented on web-rdfs/source/http_handlers.cc:
```diff
@@ -77,25 +99,30 @@ void get_handler(std::shared_ptr<HttpsServer::Response> response,
   }
 }
 
+void not_found_handler(std::shared_ptr<HttpsServer::Response> response,
```
I don't think this is necessary anymore; there was a patch by @vrnaz to handle this in #201.

LINKIWI commented on web-rdfs/source/http_handlers.cc:
```diff
@@ -77,25 +99,30 @@ void get_handler(std::shared_ptr<HttpsServer::Response> response,
-  create_file_handler(response, request);
 }
 
 void put_handler(std::shared_ptr<HttpsServer::Response> response,
```
Can you rebase? This seems out-of-date.

riceluxs1t committed:
Feature/ectests3 (#188)

* manually testing copyFromLocal with EC...fixed some bugs, changed some logs

* fixed some bugs, still debugging add_block

* fixed ec populated logic. reverted teh wrong "fix". removed the wrong comment.

* added the test cases for get erasurecoding policy in the EC case.

* unit tests for set erasure coding policy path added.

* test cases added for the file info api.

* Fixed typo.

* check parent directory for EC policy

* added test for find_parent function.

* minor logic change

* it works!

* added some comment.

* made a bug fix in set erasure coding policy.

* manually testing copyFromLocal with EC...fixed some bugs, changed some logs

* fixed some bugs, still debugging add_block

* fixed ec populated logic. reverted teh wrong "fix". removed the wrong comment.

* added the test cases for get erasurecoding policy in the EC case.

* unit tests for set erasure coding policy path added.

* test cases added for the file info api.

* Fixed typo.

* check parent directory for EC policy

* added test for find_parent function.

* minor logic change

* it works!

* added some comment.

* check parent directory for EC policy

* added test for find_parent function.

* minor logic change

* it works!

* added some comment.

* testing stuff

* lint error fix.

* whoops

* or packet empty

* added create ops to add_Block_group

* wtf is going on

* added EC RW test

* set state to enabled.

* ec rwtest updated

* added get block locations

* update test

* fix test

* fixed EC get Block Locations

* get_file_size fix for EC

* added some file_length EC stuff

* ugh

* test

* small fix to zk nn client complete

* lint fixes and modified the getfileinfo unit test so that it does not affect subsequent unit tests.

* ec file length fix (hacky version)

jsu100 committed:
Feature/file append jackson (#214)

* Added in append_file and helpers code without changing get_block_locations

* uploaded correct file

* updated without append_file, but has some helpers

* updated without append_file, but has some helpers (reverted from commit ab65b5df43f93b3a34262d9fb8a0236408655efd)

* fixed some style errors

* still fixing style errors

* fixing more style errors

* styling errors pt2

* styling errors 3

* styling errors 4

* added todo comments to append_file

* fixe dsome word wrappers!

* hopefully fixing the build error!

* Fixing errors with merging develop

* more merge issues

* more merge errors

* fixed more errors

* merge conflicts fixing

* fixed lots of the append file code!!!!!!! -mjp

* fixed some proto errors!

* attempt with *

* trying &

* did some const magic

* impl

* style!

* Initial commit for AppendFileTest

* Added simple append test

* Added simple append test

* called complete at end of get_primary_block_info

* testing test hypothesis

* this should work.

* fixed copyright and new line. should work.

* should fail.

* fixed test back, and added stuff in cmakelsits! (also roomeved exists line) (should run)

* fixed baby error.

* changed test name! should text!

* wkg on tests. dealing with some privacy issues. going to ask aj tomorrow.

* added AppendFileTest, linked in CMakeLists

* added initializeDatanodes to top of AppendFileTest (copied from other Integration Tests)

* added necessary imports

* working on AppendTest. Trying to fix privacy issue by replacing Zookeeperfilepath with physical path

* fixed style error in code provided by testing group

* fix to run tests

* style

* appendTest changed path from client->leases to /leases/

* comments

* Added some review comments on recently added name node code.

* should work.

* should fail.

* SHOULD ACTUALLY FAIL.

* Added test 2. Should pass!

* should pass. added 3rd test!

* Updated review comments. Removed extraneous code, and got rid of incorrect call to complete in get_primary_block_data.

* added in some of p's changes!!!

* Add AppendFileTest to CMake so that it can be built

* add case for appending to a nonexistent file

* Remove files created by the tests

* Add a test that appends to a file on a separate client and make sure that the append changes is synchronized in another client

* started renewlease changes for check_lease

* get rid of unneeded variables in test

* Add test for when mulitple clients append to separate files

* Add test with "N" threads that append to separate files

* Add test with 2 threads that attempt to append to the same file

* modified check_lease

* style

* fixed errors

* Delete redundant dependency in CMakeList

* adding back end to end test for append

* Fixed bugs in AppendFileTest

* Code improvements and further bug fixes in AppendFileTest

* Test ReadWrite

* Add correctness case for when two clients append to the same file.

* Fix lint errors

* Filter out non simple append tests

* edited out ReadWrite Tests again

* fixed style errors

* Fix the simple test.

* print out hadoop version.

* Print out hadoop version.

* Comment out the test.

* Fix build error.

* aaaaa fix lint error

Commit:
Feature/ectests4 (#218)

* fixed the partial stripe read error

* fix get_block_size bug

* lints

* lint again

* remove useless example test from StorageTest - why so much zkconnectionloss?? :(

willkoh committed:
Fixed error on check_ack that errors on all of them if one breaks. (#196)

* Fixed error on check_ack that errors on all of them if one breaks.

* Return correct error value.

* Fixed an error of not recursively deleting.

vnazz committed:
Feature/webrdfs append permis (#216)

* Add append and set permission skeleton

* Fix post handler response

LINKIWI committed:
Frontend handler for serving static client file (#219)

* Frontend handler for serving static client file

* Add simple test

* Add index.html placeholder

* Lint fix

heliumj committed:
remove 404

heliumj committed:
rebase and fix ls

heliumj committed:
Merge branch 'feature/webrdfs-create-ls' of https://github.com/comp413-2017/RDFS into feature/webrdfs-create-ls

heliumj committed:
fix tests after vagrant provision

LINKIWI commented:
Are you sure this is merged correctly? The changeset is huge and includes a lot of unrelated commits

LINKIWI commented on test/webRDFS/WebServerTest.cc:
```diff
@@ -42,9 +42,86 @@ TEST(WebServerTest, testRead) {
+                           "dirToLs?op=LISTSTATUS > actualResultLs"));
+
+  // Check that results match except for the access times
+  ASSERT_EQ(256,
```
This seems strange, why do you assert that the expected and actual results are completely different?
